playwright/src/main/java/com/microsoft/playwright/impl/UrlMatcher.java

```diff
@@ -77,21 +78,32 @@ static String resolveUrl(URL baseUrl, String spec) {
   }
 
   private UrlMatcher(URL baseURL, String glob, Pattern pattern, Predicate<String> predicate) {
-    this.baseURL = baseURL;
+    this.baseURL = baseURL != null ? baseURL.toString() : null;
     this.glob = glob;
     this.pattern = pattern;
     this.predicate = predicate;
   }
 
   boolean test(String value) {
+    return testImpl(baseURL, pattern, predicate, glob, value);
+  }
+
+  private static boolean testImpl(String baseURL, Pattern pattern, Predicate<String> predicate, String glob, String value) {
     if (pattern != null) {
       return pattern.matcher(value).find();
     }
     if (predicate != null) {
       return predicate.test(value);
     }
     if (glob != null) {
-      return Pattern.compile(globToRegex(resolveUrl(baseURL, glob))).matcher(value).find();
+      if (!glob.startsWith("*")) {
+        // Allow http(s) baseURL to match ws(s) urls.
+        if (baseURL != null && Pattern.compile("^https?://").matcher(baseURL).find() && Pattern.compile("^wss?://").matcher(value).find()) {
+          baseURL = baseURL.replaceFirst("^https?", "ws");
```

Comment:
This will replace both `http` and `https` with `ws`. I assume we want to replace https->wss and http->ws, don't we?